ops/charm.py

```diff
@@ -1896,6 +1922,29 @@ def __init__(self, name: str, raw: Optional[Dict[str, Any]] = None):
         self.additional_properties = raw.get('additionalProperties', True)
 
 
+class ConfigMeta:
+    """Object containing metadata about a config option."""
+
+    name: str
+    """Name of the config option."""
+
+    type: Literal['string', 'int', 'float', 'boolean', 'secret']
+    """Type of the config option."""
+
+    default: Any
```

Comment:
Could/should this be `Union[str, int, float, bool, None]`?

Reply:
Nice catch, yes it should.

```suggestion
    default: Union[bool, int, float, str, None]
```